### ISSUE TITLE
Race Condition

### ISSUE THREAD

**Issue by boyter** [boyter/scc#116]
Seems to be related to cuba, going to raise over there as well

```
$ go run -race .
==================
WARNING: DATA RACE
Write at 0x00c0000200e0 by goroutine 77:
  sync/atomic.AddInt32()
      /usr/local/go/src/runtime/race_amd64.s:269 +0xb
  github.com/boyter/scc/vendor/github.com/dbaggerman/cuba.(*Pool).runWorker()
      /home/bboyter/Go/src/github.com/boyter/scc/vendor/github.com/dbaggerman/cuba/pool.go:119 +0x52

Previous read at 0x00c0000200e0 by goroutine 9:
  github.com/boyter/scc/vendor/github.com/dbaggerman/cuba.(*Pool).PushAll()
      /home/bboyter/Go/src/github.com/boyter/scc/vendor/github.com/dbaggerman/cuba/pool.go:79 +0x112
  github.com/boyter/scc/vendor/github.com/dbaggerman/cuba.(*Handle).Sync()
      /home/bboyter/Go/src/github.com/boyter/scc/vendor/github.com/dbaggerman/cuba/handle.go:18 +0x8b
  github.com/boyter/scc/vendor/github.com/dbaggerman/cuba.(*Pool).runWorker()
      /home/bboyter/Go/src/github.com/boyter/scc/vendor/github.com/dbaggerman/cuba/pool.go:131 +0x108

Goroutine 77 (running) created at:
  github.com/boyter/scc/vendor/github.com/dbaggerman/cuba.(*Pool).PushAll()
      /home/bboyter/Go/src/github.com/boyter/scc/vendor/github.com/dbaggerman/cuba/pool.go:83 +0xe2
  github.com/boyter/scc/vendor/github.com/dbaggerman/cuba.(*Handle).Sync()
      /home/bboyter/Go/src/github.com/boyter/scc/vendor/github.com/dbaggerman/cuba/handle.go:18 +0x8b
  github.com/boyter/scc/vendor/github.com/dbaggerman/cuba.(*Pool).runWorker()
      /home/bboyter/Go/src/github.com/boyter/scc/vendor/github.com/dbaggerman/cuba/pool.go:131 +0x108

Goroutine 9 (running) created at:
  github.com/boyter/scc/vendor/github.com/dbaggerman/cuba.(*Pool).Push()
      /home/bboyter/Go/src/github.com/boyter/scc/vendor/github.com/dbaggerman/cuba/pool.go:63 +0x188
  github.com/boyter/scc/processor.(*DirectoryWalker).Walk()
      /home/bboyter/Go/src/github.com/boyter/scc/processor/file.go:89 +0x188
  github.com/boyter/scc/processor.Process.func1()
      /home/bboyter/Go/src/github.com/boyter/scc/processor/processor.go:357 +0xe9
==================
───────────────────────────────────────────────────────────────────────────────
Language                 Files     Lines   Blanks  Comments     Code Complexity
───────────────────────────────────────────────────────────────────────────────
Java                        23      3912      798       651     2463        547
Go                          20      5562      764       235     4563        798
Python                       8       343       18        19      306         29
```

Going to add test in test-all.sh file to ensure this is caught

**Comment by boyter** [comment]
Reported as fixed in cuba 0.3.1. Pending testing.

**Comment by boyter** [comment]
Resolved.
